src/SpotifySongSearch.jsx

```diff
@@ -40,46 +43,122 @@ const SpotifySongSearch = ({ authorized }) => {
-          .map((track) => track.trackUri)}
+    <>
+      <SearchForm
+        handleClick={handleClick}
+        handleTracksChange={handleTracksChange}
+        rawText={rawText}
       />
+      <ResponsesContainer isFetching={isFetching} responses={responses} />
+      <CreatePlaylistForm responses={responses} />
+    </>
+  );
+};
+
+const ResponseType = PropTypes.shape({
+  trackUri: PropTypes.string,
+  albumCover: PropTypes.string,
+  artists: PropTypes.arrayOf(PropTypes.string),
+  album: PropTypes.string,
+  name: PropTypes.string,
+  duration: PropTypes.number,
+  hasError: PropTypes.bool,
+});
+
+const CreatePlaylistForm = ({ responses }) => {
+  return (
+    <div className="wrapper spotlight style3">
+      <div className="inner">
+        <CreatePlaylistButton
+          tracksUris={responses
+            .filter((response) => !response.hasError)
+            .map((track) => track.trackUri)}
+        />
+      </div>
     </div>
   );
 };
 
-SpotifySongSearch.propTypes = {
-  authorized: PropTypes.bool,
+CreatePlaylistForm.propTypes = {
+  responses: PropTypes.arrayOf(ResponseType),
+};
+
+CreatePlaylistForm.defaultProps = {
+  responses: [],
+};
+
+const ResponsesContainer = ({ isFetching, responses }) => {
+  return (
+    <div className="wrapper alt spotlight style2">
+      <div className="inner">
+        <div className="sectionContentContainer">
+          {isFetching ? (
+            <h3 className="fullWidthText">Fetching...</h3>
+          ) : (
+            <ResponsesView responses={responses} />
+          )}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+ResponsesContainer.propTypes = {
+  isFetching: PropTypes.bool,
+  responses: PropTypes.arrayOf(ResponseType),
+};
+
+ResponsesContainer.defaultProps = {
+  isFetching: false,
+  responses: [],
+};
+
+const SearchForm = ({ handleTracksChange, rawText, handleClick }) => {
+  const { authorized } = useContext(UserContext);
+  return (
+    <section className="wrapper spotlight style3 ">
+      <div className="inner">
+        <section className="sectionContentContainer">
+          <h3 className="major fullWidthText">Enter track list</h3>
+          <textarea
+            className="searchBox"
+            placeholder='Enter you playlist items in "artist - title" format'
+            onChange={handleTracksChange}
+            value={rawText}
+            disabled={!authorized}
+          />
+          <button
+            className="searchButton button primary"
+            type="button"
+            onClick={handleClick}
+            disabled={!authorized}
+          >
+            Search
+          </button>
+        </section>
+      </div>
+    </section>
+  );
+};
+
+SearchForm.propTypes = {
+  handleTracksChange: PropTypes.func,
+  rawText: PropTypes.string,
+  handleClick: PropTypes.func,
 };
 
-SpotifySongSearch.defaultProps = {
-  authorized: false,
+SearchForm.defaultProps = {
+  handleTracksChange: () => {},
```

Comment:
do you really need all 3 props as optional with default initializers? it looks like it could be required without any default props at all

Reply:
I have removed all defaultProps, bcause I do not need them, but linter qeruired them.